src/monodroid/jni/monodroid-glue.cc

```diff
@@ -98,6 +99,7 @@ using namespace xamarin::android::internal;
 // implementation details as it would prevent mkbundle from working
 #include "mkbundle-api.h"
 
+std::mutex MonodroidRuntime::java_reflection_lock;
```

Comment:
"naming & scope & semantics, oh my!": JNI *itself* is thread-safe.  (The target Java class?  Possibly not.  Our usage?  Possibly not.  JNI *itself* certainly is thread-safe.)

My problem with the *name* `java_reflection_lock` is that it *implies* that we need to lock "all" JNI access, or all "Java reflection" access (whatever that would be?).  This isn't the case; it's only used to protect *setting* our `MonodroidRuntime::java_TimeZone` field.

We could go "very granular" and have a `java_TimeZone_lock` field, or we could have a `lazy_jni_lookup_lock` lock for use with `MonodroidRuntime::java_TimeZone` and other "lazily looked up" fields in the future, or some other better name.

`java_reflection_lock` is too easily misinterpreted.

Reply:
> "naming & scope & semantics, oh my!": JNI _itself_ is thread-safe. (The target Java class? Possibly not. Our usage? Possibly not. JNI _itself_ certainly is thread-safe.)
> 
The lock protects our class field, we can't avoid using it, alas.

> My problem with the _name_ `java_reflection_lock` is that it _implies_ that we need to lock "all" JNI access, or all "Java reflection" access (whatever that would be?). This isn't the case; it's only used to protect _setting_ our `MonodroidRuntime::java_TimeZone` field.
> 
> We could go "very granular" and have a `java_TimeZone_lock` field, or we could have a `lazy_jni_lookup_lock` lock for use with `MonodroidRuntime::java_TimeZone` and other "lazily looked up" fields in the future, or some other better name.
> 
> `java_reflection_lock` is too easily misinterpreted.

I'll rename it